lib/mongoid/contextual/mongo.rb

```diff
@@ -777,6 +780,17 @@ def third_to_last!
         third_to_last || raise_document_not_found_error
       end
 
+      # Schedule a task to load documents for the context.
+      #
+      # Depending on the Mongoid configuration, the scheduled task can be executed
+      # immediately on the caller's thread, or can be scheduled for an
+      # asynchronous execution.
+      #
+      # @api private
+      def load_async
+        @documents_loader = DocumentsLoader.new(view, klass, criteria) unless @documents_loader
```

Comment:
is there a reason you didn't just use the `||=` operator here? Do you want it to return `nil` when `@documents_loader` already exists?

Reply:
Good catch, thanks. Fixed.